core/unix/os.c

```diff
@@ -8765,11 +8765,24 @@ get_dynamo_library_bounds(void)
     check_start = dynamo_dll_start;
     dynamorio_libname = IF_UNIT_TEST_ELSE(UNIT_TEST_EXE_NAME, DYNAMORIO_LIBRARY_NAME);
 #    endif /* STATIC_LIBRARY */
+
+#    if !defined(STATIC_LIBRARY) && defined(LINUX)
+    static char dynamorio_libname_buf[MAXIMUM_PATH];
+    res = memquery_library_bounds(NULL, &check_start, &check_end, dynamorio_library_path,
+                                  BUFFER_SIZE_ELEMENTS(dynamorio_library_path),
+                                  dynamorio_libname_buf,
+                                  BUFFER_SIZE_ELEMENTS(dynamorio_libname_buf));
+    dynamorio_libname = IF_UNIT_TEST_ELSE(UNIT_TEST_EXE_NAME, dynamorio_libname_buf);
+#    else
     res = memquery_library_bounds(dynamorio_libname, &check_start, &check_end,
                                   dynamorio_library_path,
-                                  BUFFER_SIZE_ELEMENTS(dynamorio_library_path));
+                                  BUFFER_SIZE_ELEMENTS(dynamorio_library_path), NULL, 0);
+#    endif
+
     LOG(GLOBAL, LOG_VMAREAS, 1, PRODUCT_NAME " library path: %s\n",
         dynamorio_library_path);
+    LOG(GLOBAL, LOG_VMAREAS, 1, PRODUCT_NAME " library file path: %s\n",
+        dynamorio_library_filepath);
```

Comment:
Shouldn't be printed until it's set below!

Reply:
Thanks.